packages/plugin-ext/src/main/browser/view-column-service.ts

```diff
@@ -0,0 +1,96 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject } from 'inversify';
+import { Emitter, Event } from '@theia/core';
+import { ApplicationShell } from '@theia/core/lib/browser/shell/application-shell';
+import { TheiaDockPanel } from '@theia/core/lib/browser/shell/theia-dock-panel';
+import { toArray } from '@phosphor/algorithm';
+import { Widget } from '@phosphor/widgets';
+
+@injectable()
+export class ViewColumnService {
+    private mainPanel: TheiaDockPanel;
+    private columnValues = new Map<string, number>();
+    private viewColumnIds = new Map<number, string[]>();
+
+    protected readonly onViewColumnChangedEmitter = new Emitter<{ id: string, viewColumn: number }>();
+
+    constructor(
+        @inject(ApplicationShell) private readonly shell: ApplicationShell,
+    ) {
+        this.mainPanel = this.shell.mainPanel;
+
+        let oldColumnValues = new Map<string, number>();
+        const update = async () => {
+            await new Promise((resolve => setTimeout(() => resolve())));
+            this.updateViewColumns();
+            this.viewColumnIds.forEach((ids: string[], viewColumn: number) => {
+                ids.forEach((id: string) => {
+                    if (!oldColumnValues.has(id) || oldColumnValues.get(id) !== viewColumn) {
+                        this.onViewColumnChangedEmitter.fire({ id, viewColumn });
+                    }
+                });
+            });
+            oldColumnValues = new Map(this.columnValues.entries());
+        };
+        this.mainPanel.widgetAdded.connect(() => update());
+        this.mainPanel.widgetRemoved.connect(() => update());
+    }
+
+    get onViewColumnChanged(): Event<{ id: string, viewColumn: number }> {
+        return this.onViewColumnChangedEmitter.event;
+    }
+
+    updateViewColumns(): void {
+        const positionIds = new Map<number, string[]>();
+        toArray(this.mainPanel.widgets()).forEach((widget: Widget) => {
```

Comment:
Should not it be based on tab groups (i.e. tab group === view position), not widgets.

Reply:
fixed